lib/ansible/modules/cloud/cloudstack/cs_instance.py

```diff
@@ -743,11 +749,33 @@ def update_instance(self, instance, start_vm=True):
 
         security_groups_changed = self.security_groups_has_changed()
 
+        # Volume data
+
+        args_volume_update = {}
+        root_disk_size = self.module.params.get('root_disk_size')
+        root_disk_size_changed = False
+
+        if root_disk_size is not None:
+            res = self.query_api('listVolumes', type='ROOT', virtualmachineid=instance['id'])
+            [volume] = res['volume']
+
+            size = volume['size'] >> 30
+
+            args_volume_update['id'] = volume['id']
+            args_volume_update['size'] = root_disk_size
+
+            shrinkok = self.module.params.get('allow_root_disk_shrink')
+            if shrinkok:
+                args_volume_update['shrinkok'] = shrinkok
+
+            root_disk_size_changed = root_disk_size != size
```

Comment:
one more question, this would also be "true" for enlargement of the disk, would this work as expected? if yes, `allow_root_disk_shrink` would be a misleading name because it would not only resize to lower, but also resize to bigger volume --> `allow_root_disk_resize`

Reply:
it's really only for _shrinkage` that a special flag needs to be sent. `allow_root_disk_resize` would be misleading as the nothing special is required for enlarging a disk.

http://cloudstack.apache.org/api/apidocs-4.4/user/resizeVolume.html

Reply:
@resmo the shrink option is only for changing the disk size to a smaller partition. It requires more actions from the user if it should success:
- https://github.com/apache/cloudstack/blob/756a7e89cbcec7a1aa98b3f470cbdf8181af75ad/server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java#L1022
- http://libguestfs.org/virt-resize.1.html (section "SHRINKING A VIRTUAL MACHINE DISK")

Reply:
so, that means, we should change this condition to the following, right?
 ~~~python
root_disk_size_changed = root_disk_size < size
~~~

Reply:
No, you can resize the volume up or down, ***but*** if you want to resize it down you must specify the flag `shrink` to ensure you know what you are doing since it requires some preparation of the volume.

Reply:
ok, got it! :+1: